.github/workflows/merlin.yml

```diff
@@ -0,0 +1,433 @@
+name: Merlin CI Workflow
+on:
+  push:
+    branches: 
+    - main 
+    - v1beta1  # TODO: remove v1beta1
+  pull_request:
+    branches: 
+    - main 
+    - v1beta1  # TODO: remove v1beta1
+
+env:
+  ARTIFACT_RETENTION_DAYS: 7
+
+jobs:
+  create-version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.create_version.outputs.version }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: create_version
+        name: Create version string
+        run: |
+          # Strip git ref prefix from version
+          VERSION=$(git describe --tags --always --first-parent)
+
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "master" ] && VERSION=latest
+
+          echo "::set-output name=version::${VERSION}"
+
+  test-batch-predictor:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - uses: actions/cache@v3
+        with:
+          path: ~/.local/share/virtualenvs
+          key: ${{ runner.os }}-python-${{ steps.setup-python.outputs.python-version }}-pipenv-${{ hashFiles('Pipfile.lock') }}
+      - name: Install dependencies
+        working-directory: ./python/batch-predictor
+        run: |
+          pip install pipenv
+          make setup
+      - name: Run batch-predictor test
+        working-directory: ./python/batch-predictor
+        run: make unit-test
+
+  test-pyfunc-server:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.7.9
```

Comment:
Why is this 3.7.9 instead of 3.7 like others?

Reply:
No particular reason. Let me try updating it.